packages/mobile/src/app/App.tsx

```diff
@@ -45,71 +46,30 @@ BigNumber.config({
   },
 })
 
-interface State {
-  reactInitTime?: number
-  reactLoadTime?: number
-}
-
 // Enables LayoutAnimation on Android. It makes transitions between states smoother.
 // https://reactnative.dev/docs/layoutanimation
 // Disabling it for now as it seems to cause blank white screens on certain android devices
 // if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
 //   UIManager.setLayoutAnimationEnabledExperimental(true)
 // }
 
-const createEventListener: (
-  eventName: string,
-  callback: (reactInitTime: string) => void
-) => EmitterSubscription | null = (
-  eventName: string,
-  callback: (reactInitTime: string) => void
-): EmitterSubscription | null => {
+function getEventEmitter() {
   if (Platform.OS === 'android') {
-    return new NativeEventEmitter().addListener(eventName, callback)
+    return new NativeEventEmitter()
+  } else {
+    // ios
+    return new NativeEventEmitter(AnalyticsManager)
   }
-  // TODO: Add listener for iOS
-  return null
 }
 
 export class App extends React.Component {
-  state: State = {
-    reactInitTime: undefined,
-    reactLoadTime: undefined,
-  }
-
-  appStartListener: EmitterSubscription | null = createEventListener(
-    'AppStartedLoading',
-    (reactInitTime: string) => {
-      this.setState({
-        reactInitTime: +reactInitTime,
-        reactLoadTime: Date.now(),
-      })
-    }
-  )
+  reactLoadTime = Date.now()
 
   async componentDidMount() {
     await ValoraAnalytics.init()
-    const { width, height } = Dimensions.get('window')
-
-    let reactLoadDuration
-    let appLoadDuration
-
-    if (this.state.reactInitTime && this.state.reactLoadTime) {
-      const appLoadedTime = Date.now()
-      reactLoadDuration = (this.state.reactLoadTime - this.state.reactInitTime) / 1000
-      appLoadDuration = (appLoadedTime - this.state.reactInitTime) / 1000
-    }
-
-    ValoraAnalytics.startSession(AppEvents.app_launched, {
-      deviceHeight: height,
-      deviceWidth: width,
-      reactLoadDuration,
-      appLoadDuration,
-    })
 
-    if (this.appStartListener) {
-      this.appStartListener.remove()
-    }
+    const appLoadedTime = Date.now()
```

Comment:
Instead of having this as an input for `getAppStartTimeFromNative`, can we instead just define it inside the function? I think it would be helpful to contain as much of the variables and logic within that function as possible